### DISCUSSION TITLE
Can the "crawler" project be opened up?

### DISCUSSION THREAD

**Discussion by CodeMan99** [devcontainers/community#86] [Q&A]
I would like to contribute to the _crawler_ tool that transforms the [collection-index.yml](https://github.com/devcontainers/devcontainers.github.io/blob/gh-pages/_data/collection-index.yml) into the [OCI Registry Index](https://github.com/orgs/devcontainers/packages/container/package/index).

The primary idea I would like to explore is validation of `devcontainer-feature.json` and `devcontainer-template.json` files when a PR is opened against _collection-index.yml_. This would help ensure the schema is respected.

**Comment by samruddhikhandale** [comment]
Hi 👋 

Currently, the crawler is private and we haven't yet decided to make it public.

> The primary idea I would like to explore is validation of devcontainer-feature.json and devcontainer-template.json files when a PR is opened against collection-index.yml. This would help ensure the schema is respected.

This is a super sweet idea, and would definitely save maintainer's time in approving PRs. ✨ 

Can we add this validation to a new GitHub action which runs automatically for the PRs which updates `collection-index.yml`? This will be needed in https://github.com/devcontainers/devcontainers.github.io repository

**Reply by chrmarti** [reply]
Do we validate the `devcontainer-feature.json` and `devcontainer-template.json` in the https://github.com/devcontainers/action? Maybe that could be reused. /cc @joshspicer
